pkg/config/creds.go

```diff
@@ -18,6 +19,10 @@ type Creds struct {
 	AppKey string
 }
 
+const credsKey = "creds"
+
+var credsCache sync.Map
```

Comment:
The use of a `Map` seems a little bit overkill here since it will ever contain only a single element.
You replace it by:
```suggestion
var credsCache Creds
var sync.Mutex CredsMutex
```
I agree that in this case, we’ll have to handle a Mutex in `cacheCreds`, `getCredsFromCache` and `resetCredsCache`.